container/container.go

```diff
@@ -946,23 +948,27 @@ func (b *ContainerWorker) ResourcesCreate(agreementId string, agreementProtocol
 		return endpoints
 	}
 
-	workloadRWStorageDir := b.workloadStorageDir(agreementId)
+	workloadRWStorageDir, useVolume := b.workloadStorageDir(agreementId)
 
-	// create RO workload storage dir if it doesnt already exist
-	if err := os.Mkdir(workloadRWStorageDir, 0700); err != nil {
-		if pErr, ok := err.(*os.PathError); ok {
-			if pErr.Err.Error() != "file exists" {
+	if !useVolume {
+		// create RO workload storage dir if it doesnt already exist
+		if err := os.Mkdir(workloadRWStorageDir, 0700); err != nil {
+			if pErr, ok := err.(*os.PathError); ok {
+				if pErr.Err.Error() != "file exists" {
+					return nil, err
+				}
+			} else {
 				return nil, err
 			}
-		} else {
-			return nil, err
 		}
-	}
 
-	glog.V(5).Infof("Writing raw config to file in %v. Config data: %v", workloadRWStorageDir, string(configureRaw))
-	// write raw to workloadRWStorageDir
-	if err := ioutil.WriteFile(path.Join(workloadRWStorageDir, "Configure"), configureRaw, 0644); err != nil {
-		return nil, err
+		glog.V(5).Infof("Writing raw config to file in %v. Config data: %v", workloadRWStorageDir, string(configureRaw))
+		// write raw to workloadRWStorageDir
+		if err := ioutil.WriteFile(path.Join(workloadRWStorageDir, "Configure"), configureRaw, 0644); err != nil {
+			return nil, err
+		}
+	} else {
+		// do we need to create the volume here?
```

Comment:
Where do you create the volume?

Reply:
I need to remove the `else` clause here. For the volume case, we do nothing here. The volume is created by docker as long as the source in `source:dest:option` in the HostConfig.Binds is a name instead of a file system path and the named volume does not exist.